ring_benchmark/ring.erl

```diff
@@ -0,0 +1,43 @@
+-module(ring).
+-export([main/1]).
+
+node_loop() ->
+  receive
+    {msg, []} -> ok;
+    {msg, [FirstNode | OtherNodes]} ->
+      FirstNode ! {msg, OtherNodes},
+      node_loop()
+  end.
+    
+
+% N processes, M messages
+main([NArg,MArg]) ->
+  N = arg_to_number(NArg),
+  M = arg_to_number(MArg),
+  [FirstNode | Nodes] = create_processes(N, M),
+  BeforeFistMessage = os:timestamp(),
+  FirstNode ! {msg, Nodes},
+  receive
+    {msg, _} -> ok
+  end,
+  AfterLastMessage = os:timestamp(),
+  ElapsedTime = timer:now_diff(AfterLastMessage, BeforeFistMessage),
+  io:format("Processes: ~p, Messages ~p in ~pms~n", [N, M, ElapsedTime]).
+
+create_processes(N, M) ->
+  Processes = [spawn_link(fun () -> node_loop() end)
+               || _ <- lists:seq(1, N)],
+  ProcessList =
+    lists:foldl(fun(X,Acc) ->
+                    Acc ++ X
+                end,[],lists:duplicate(M, Processes)),
+  CompleteLoop = ProcessList ++ [self()],
```

Comment:
why dont you pass self as an argument instead of adding it to the process list? (because they sort of have a different nature)